src/transport/tls/listen.rs

```diff
@@ -172,14 +182,15 @@ impl<L: HasConfig, G> Listen<L, G> {
             });
 
             incoming
-                .map(move |(socket, remote_addr)| {
+                .map(move |(mut socket, remote_addr)| {
```

Comment:
As I mentioned above, I don't think we need a mut reference to set keepalive, so this shouldn't be necessary as far as I can tell?